test/modules/ssh-docker-integration.ts

```diff
@@ -0,0 +1,489 @@
+sudo apt-get update
+sudo apt-get install -y ca-certificates curl gnupg
+
+sudo mkdir -m 0755 -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+
+echo "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+sudo apt-get update
+sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+sudo groupadd docker
+sudo usermod -aG docker ubuntu
+newgrp docker'
+          FROM subnet
+          WHERE availability_zone = '${availabilityZone1}'
+          LIMIT 1;
+
+        INSERT INTO instance_security_groups (instance_id, security_group_id) SELECT
+          (SELECT id FROM instance WHERE tags ->> 'name' = 'ssh-${prefix}-1'),
+          (SELECT id FROM security_group WHERE group_name='ssh-${prefix}-sg' AND region = '${region}');
+      COMMIT;
+    `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    )((e?: any) => {
+      if (!!e) return done(e);
+      done();
+    });
+  });
+
+  it('commits the transaction', commit());
+
+  it('installs the ssh_accounts and aws_ec2_metadata modules', install(['ssh_accounts', 'aws_ec2_metadata']));
+
+  itDocs('adds a new ssh server', (done: (e?: Error) => any) => {
+    query(
+      `
+          INSERT INTO ssh_credentials ("name", hostname, username, private_key)
+          VALUES ('${prefix}', (SELECT host(im.public_ip_address)
+                                FROM instance_metadata im
+                                         INNER JOIN instance i ON im.instance_id = i.instance_id
+                                WHERE i.tags ->> 'name' = 'ssh-${prefix}-1'
+                                LIMIT 1), 'ubuntu', $$${privateKey}$$);
+      `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    )((e?: any) => {
+      if (!!e) return done(e);
+      done();
+    });
+  });
+
+  itDocs(
+    'waits for cloud-init to finish',
+    query(
+      `
+    SELECT * FROM ssh_exec('${prefix}', 'cloud-init status --wait');
+  `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  itDocs(
+    'checks docker is running',
+    query(
+      `
+    SELECT * FROM ssh_exec('${prefix}', 'sudo docker ps');
+  `,
+      (res: any[]) => expect(res[0].stdout.trim()).toContain('CONTAINER ID'),
+      true,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('installs the ssh_docker modules', install(['ssh_docker']));
+
+  it('starts a transaction', begin());
+  itDocs(
+    'creates a new httpd container',
+    query(
+      `
+          INSERT INTO docker_container (server_name, name, image, ports, state)
+          VALUES ('${prefix}', 'httpd-container', 'httpd',
+                  '{"80/tcp": [{"HostIp": "", "HostPort": "8080"}]}', 'created');
```

Comment:
what happens if we create the container without defining the `state`?

Reply:
Yes. A `cloud.create` will create it in the `CREATED` state, then a subsequent `cloud.update` will start it.

Reply:
@mtp1376 he was asking what happens if you *don't* set the `state` parameter, so I told him that it [defaults to `running`](https://github.com/iasql/iasql/pull/2374/files#diff-5ecdbef8c988b5c5dda58c5acec1505812ddd51348015ca5b234ffba947fd7edR137) and so it should immediately start the docker image on the other side.

Reply:
Yes your explanation is right, but the cloud.create will just create the container and then a cloud.update will start it (because the cloud state is not exactly like local: created vs running).

Reply:
Oh? So it actually always goes through that two-step process?

Reply:
Yes. It'll go through a two step process currently.